python/lang/security/deserialization/avoid-jsonpickle.yaml

```diff
@@ -1,32 +1,147 @@
 rules:
 - id: avoid-jsonpickle
-  patterns:
-  - pattern: |
-      jsonpickle.decode($PAYLOAD,...)
-  - pattern-not: |
-      jsonpickle.decode("...",...)
+  mode: taint
+  pattern-sources:
+      - pattern-either:
+          - patterns:
+              - pattern: |
+                  def $HANDLER($EVENT, context):
+                    ...
+              - focus-metavariable: $EVENT
+          - patterns:
+              - pattern-either:
+                  - pattern-inside: |
+                      @rest_framework.decorators.api_view(...)
+                      def $FUNC(..., $REQ, ...):
+                        ...
+                  - patterns:
+                      - pattern-either:
+                          - pattern-inside: >
```

Comment:
`- pattern-inside: |` for consistency? also spacing for the class VIEW -- could all be in one line